odoo/tools/misc.py

```diff
@@ -1641,3 +1641,21 @@ def hmac(env, scope, message, hash_function=hashlib.sha256):
         message.encode(),
         hash_function,
     ).hexdigest()
+
+
+bots = ["bot", "crawl", "slurp", "spider", "curl", "wget", "facebookexternalhit", "whatsapp", "trendsmapresolver", "pinterest", "instagram"]
+def is_a_bot(user_agent):
```

Comment:
`odoo.tools.misc` seems like a pessimal place for this function, why not leave it where it was, or as a utlity function in `odoo.http`?

Converting it to a function also precludes overriding the method to add more filters, which was a feature of the previous location.

Reply:
Hi,
Thanks for your comments. I moved the function in `odoo.http`. While it is true that the method can't be overwritten now, I can't think of a reason why this method would need to be overwritten.

The method is trying to identify bots based on user_agent, so if a new user agent for a bot is found, it should just be added here. It doesn't make much sense to me to have a request that is only seen as a bot in certain modules...

Let me know what you and @JKE-be think.

Reply:
Because if you use a social network from an exotic country that Odoo don't have handled, you will be happy to be able to add it as a bot, instead to wait a long time that we never merge your PR in core code.

And because this move doesn't bring any advantage, only a move where you lose the opportunity to override it...

Reply:
This is an open source software, we provide core features that usually get extended by our community/partners to provide a layer on top of what we build, to fit their need, which to be honest, and it is normal, is not always sufficient to do what they need. (We can't support everything for everyone over the world)

They can't add code themselves in our codebase, so they extend this type of method, because the crawler in their country that no one knows is not in our codebase list of bot.

And I am pretty sure what they meant to keep it overidable was to keep it on ir.http model, not as a function.

Reply:
Alright, makes sense. I moved it to `odoo/addons/web/models/ir_http.py` after discussing it with std. Keeping it in `odoo/addons/http_routing/models/ir_http.py` wasn't really an option since `link_tracker` did not depend on it and adding a dependency didn't make much sense. Hope you agree with the current solution :)